google/generativeai/types/caching_types.py

```diff
@@ -33,19 +33,38 @@ def valid_cached_content_name(name: str) -> bool:
 
 
 class TTL(TypedDict):
+    # Represents datetime.datetime.now() + desired ttl
     seconds: int
+    nanos: int
 
 
-ExpirationTypes = Union[TTL, int, datetime.timedelta]
+class ExpireTime(TypedDict):
+    # Represents seconds of UTC time since Unix epoch
+    seconds: int
+    nanos: int
+
+
+ExpirationTypes = Union[TTL, ExpireTime, int, datetime.timedelta, datetime.datetime]
 
 
-def to_ttl(expiration: Optional[ExpirationTypes]) -> TTL:
-    if isinstance(expiration, datetime.timedelta):
-        return {"seconds": int(expiration.total_seconds())}
+def to_expiration(expiration: Optional[ExpirationTypes]) -> TTL | ExpireTime:
```

Comment:
This returns a dict(seconds=.., nanos=..) but gives no way to determine if it's meant to be a TTL or an ExpireTime? That's bad.

I think if you try to set an enpire_time in 1h, you may accidently set a ttl of 55years.

Reply:
Yeah, that would be bad

I will handle it separately.